apps/api/src/services/agent-manager.service.ts

```diff
@@ -832,13 +842,42 @@ export class AgentManager {
         params,
       );
 
-      // Get agents from repository
+      // Get competitions from repository
       const results = await findAgentCompetitions(agentId, params);
 
+      // Attach metrics to each competition
```

Comment:
if we do this:
```ts
const competitions = results.competitions.filter(
  (comp) => comp !== null,
);
const enhancedCompetitions = await Promise.all(     
  results.competitions.map(...)...
)
```
then the `null` checks via `if (!competition) ...` and `validCompetitions` aren't needed.